enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js

```diff
@@ -914,3 +867,51 @@ describeWithToken("formatting > sandboxes", () => {
     });
   });
 });
+
+function signInAsSandboxedUser() {
+  cy.log("**-- Logging in as sandboxed user --**");
+  cy.request("POST", "/api/session", {
+    username: sandboxed_user.email,
+    password: sandboxed_user.password,
+  });
+}
+
+/**
+ * As per definition for `PUT /graph` from `permissions.clj`:
+ *
+ * This should return the same graph, in the same format,
+ * that you got from `GET /api/permissions/graph`, with any changes made in the wherever necessary.
+ * This modified graph must correspond to the `PermissionsGraph` schema.
+ *
+ * That's why we must chain GET and PUT requests one after the other.
+ */
+
+function updatePermissionsGraph({
+  schema = {},
+  user_group = COLLECTION_GROUP,
+  database_id = 1,
+} = {}) {
+  if (typeof schema !== "object") {
+    throw new Error("`schema` must be an object!");
+  }
+
+  cy.log("**-- Fetch permissions graph --**");
+  cy.request("GET", "/api/permissions/graph", {}).then(
+    ({ body: { groups, revision } }) => {
+      // This mutates the original `groups` object => we'll pass it next to the `PUT` request
+      groups[user_group] = {
+        [database_id]: {
+          schemas: {
+            PUBLIC: schema,
+          },
+        },
+      };
+
+      cy.log("**-- Update/save permissions --**");
+      cy.request("PUT", "/api/permissions/graph", {
+        groups,
+        revision,
+      });
+    },
+  );
+}
```

Comment:
This is probably the most important helper function for sandboxing tests.

I could've maybe made one for updating GTAP but it seems there's a bit more moving pieces there. This should be enough for now. Please let me know if you think different.